File: fibonacci4.java
import java.math.BigInteger;

public class fibonacci4 {
	
	static BigInteger fib(int n) {
		
		BigInteger a = new BigInteger("0");
		BigInteger b = new BigInteger("1");
		BigInteger c;
		
		// trivialni primer
		if(n == 0)
			return a;
		
		// iterativni izracun rezultata
		for(int i=0; i<n-1; i++) {
			c = b;
			b = b.add(a);
			a = c;
		}
		return b;
		
	}
	
	public static void main(String[] args) throws Exception {
		
		// spremenljivka n za izracun fib(n)
		int n = 10;
		
		// zacetni cas merjenja
		long startTime = System.currentTimeMillis();
		
		BigInteger rez = fib(n);
		
		// koncni cas merjenja
		long endTime = System.currentTimeMillis();
		// izracun casa merjenja
		long timeElapsed = endTime - startTime;
		
		// izpis rezultatov
		System.out.println(rez);
		System.out.println("cas: "+timeElapsed+" ms");
		
	}
	
}

File: fibonacci4.c
#include <stdio.h>
#include <gmp.h>
#include <time.h>
#include <stdlib.h>

void fib(mpz_t rez, int n) {

	// inicializacija spremenljivk
	mpz_t a, b, c;
	mpz_init (a);
	mpz_init (b);
	mpz_init (c);
	mpz_set_ui(a, 0);
	mpz_set_ui(b, 1);

	// trivialni primer
	if(n == 0) {
		mpz_set(rez, a);
		mpz_clear(a);
		mpz_clear(b);
		mpz_clear(c);
		return;
	}

	// iterativni izracun rezultata	
	for(int i=0; i<n-1; i++) {
		mpz_set(c, b);
		mpz_add(b, b, a);
		mpz_set(a, c);
	}
	
	mpz_set(rez, b);
	mpz_clear(a);
	mpz_clear(b);
	mpz_clear(c);

}

int main() {
	
	// spremenljivka n za izracun fib(n)
	int n = 10;

	struct timespec startTime, endTime;
	// zacetni cas merjenja
	clock_gettime(CLOCK_MONOTONIC_RAW, &startTime);

	mpz_t rez;
	mpz_init(rez);
	fib(rez, n);

	// koncni cas merjenja
	clock_gettime(CLOCK_MONOTONIC_RAW, &endTime);
	// izracun casa merjenja
	int timeElapsed = (endTime.tv_sec - startTime.tv_sec) * 1000 + (endTime.tv_nsec - startTime.tv_nsec) / 1000000;

	// izpis rezultatov
	mpz_out_str(stdout, 10, rez);
	printf("\n");
	printf("cas: %d ms\n", timeElapsed);

	mpz_clear(rez);

	return 0;	

}
